### ISSUE TITLE
output 3dgs(ply)

### ISSUE THREAD

**Issue by yoyoLu0521** [aigc3d/LHM-plusplus#2]
thank you for the amazing work on LHM++! I have successfully run the testing codes, and the driven video results are very impressive.

I am currently trying to export the generated 3D Gaussian Splatting model into a .ply format so that I can inspect the posed 3D geometry in external viewers like SuperSplat or Unity.
The Issue:
The exported .ply file loads fine with correct geometry and animation, but the colors appear completely wrong (it looks like a thermal or negative image, as shown in the attached screenshot).

<img width="426" height="753" alt="Image" src="https://github.com/user-attachments/assets/bd28bf32-8459-476a-8bad-821feecbc4ec" />

My Questions:

1.Is there an official or recommended pipeline/script to output correct RGB colors SH parameters for external usage?

2.Are there any pre-trained weights available (or planned to be released) that were trained with neural_renderer_input: "rgb+feats"? If so, I assume the .ply would inherently contain the base RGB colors and be viewer-friendly.

**Comment by lingtengqiu** [comment]
Thank you for your interest in our work.

In the current version, we only release the checkpoint that belongs to the model without supervision between gaussian renderer and ground-truth image. That is to say, we utilize the reconstructed Gaussian Splating model as intermediate medium, and then renderer the "features" by Gaussian Rasterizer as the conditional input for the light-weight Neural Renderer. Unfortunately,  We do not release the model that is supervised by both gaussian rendering and neural rendering images, as mentioned in the paper. 

Sorry for your confusion, but don't worry about it, we will release a new model that could achieve either  gaussian model with true color or neural rendering images. The model will be released approximately by 30 April 🙂

**Comment by KingWang569** [comment]
> Thank you for your interest in our work.
> 
> In the current version, we only release the checkpoint that belongs to the model without supervision between gaussian renderer and ground-truth image. That is to say, we utilize the reconstructed Gaussian Splating model as intermediate medium, and then renderer the "features" by Gaussian Rasterizer as the conditional input for the light-weight Neural Renderer. Unfortunately, We do not release the model that is supervised by both gaussian rendering and neural rendering images, as mentioned in the paper.
> 
> Sorry for your confusion, but don't worry about it, we will release a new model that could achieve either gaussian model with true color or neural rendering images. The model will be released approximately by 30 April 🙂

Fantastic work! I'm also really looking forward to the release of this true-color PLY model. @lingtengqiu

**Comment by lingtengqiu** [comment]
We have released checkpoints with seamless support for 3DGS .ply output. Please use the model name LHMPP-700M-SMPLX-FREE.

**Comment by yoyoLu0521** [comment]
Hi,

Thank you so much for the incredibly fast update! I tested the new model, and it works great—I can now successfully export the .ply with base RGB colors!

While inspecting the exported .ply in SuperSplat, I noticed two interesting artifacts in the base 3DGS geometry/colors (screenshots attached):

Eye region artifacts: The Gaussians around the eyes appear quite messy/dark.

<img width="334" height="630" alt="Image" src="https://github.com/user-attachments/assets/e03972c6-5b6b-4444-b5d4-4266e904fc10" />

<img width="422" height="351" alt="Image" src="https://github.com/user-attachments/assets/64782d4e-8241-4d65-a9aa-391fcc38d92c" />

Floaters/Artifacts on hair: There is an unoptimized blue Gaussian patch on the back of the head.

<img width="461" height="468" alt="Image" src="https://github.com/user-attachments/assets/d5be46e4-8cba-4ba3-9c4d-e15f8533a1dc" />

<img width="486" height="494" alt="Image" src="https://github.com/user-attachments/assets/52f469c5-0bec-4e60-bc28-609b3ccf88fd" />

I completely understand that since LHM++ relies on the DPT network to render the final perfect high-frequency details, the base 3DGS might act as a rough proxy and thus accumulate these artifacts (since the neural renderer likely masks/paints over them in the final output).

My feedback/question:
I can manually clean up the blue patch  in SuperSplat for now, but I wanted to share this observation. Do you think adding some regularization losses (like opacity penalties or applying the face_id loss partially to the base RGB output) during the GS training phase could force the base 3DGS to be cleaner in future versions?

Thanks again for the amazing support! This feature is already super helpful.

**Comment by lingtengqiu** [comment]
This artifact looks a bit strange to me. I think there may be an issue with the values in the 3DGS you exported. Let’s just write a script to make it easier for you to export the corresponding 3DGS directly.

**Comment by yoyoLu0521** [comment]
Hi,

Thank you so much for the quick response and for offering to write a dedicated script! That would be incredibly helpful.

Just to give you some context on how I obtained those artifacts, here is the exact manual process I used:

I ran scripts/test/test_app_video.py.

I modified the inference pipeline (specifically inside core/models/modeling_humana4o_lrm.py) to additionally return the GaussianModel object (i.e., render_results["3dgs"][0]).

I directly called the native .save_ply() method on this object to save the file.

Finally, I loaded the exported .ply directly into SuperSplat for visualization.

It is very likely that my direct export method missed some necessary value conversions or normalization steps for the new model, which led to those weird artifacts around the eyes and hair.

Thank you again for taking the time to write an official export script. I'm really looking forward to trying it out!

**Comment by lingtengqiu** [comment]
you did run it incorrectly. Please wait a moment; we are updating this feature right now. see：
<img width="765" height="611" alt="Image" src="https://github.com/user-attachments/assets/c96651d4-74ce-4492-82f3-f9bcb757dec3" />

**Comment by lingtengqiu** [comment]
see 
```
cd LHM-plusplus

python scripts/inference/to_gs_ply.py \
  --model_name LHMPP-700M-SMPLX-FREE \
  --pose_dir "./motion_video/BasketBall_I/smplx_params/00014.json" \
  --image_glob "./assets/example_multi_images/00000_yuliang_*.png"
```
